subprojects/frameworklauncher/yarn/src/main/java/com/microsoft/frameworklauncher/service/Service.java

```diff
@@ -303,6 +303,7 @@ private ContainerLaunchContext setupContainerLaunchContextForDefaultAM(
 
     // SetupEntryPoint
     Vector<CharSequence> vargs = new Vector<>(30);
+    vargs.add("trap ' ' TERM;"); // prevents intermediate "bash -c" interpreter from immediate exit after receiving TERM signal
     vargs.add(ApplicationConstants.Environment.JAVA_HOME.$$() + "/bin/java");
```

Comment:
Launcher is general purposed, do we have to hard code it here?

Reply:
Check the comment in [dockerContainerScript](src/rest-server/src/templates/dockerContainerScript.mustache) for explanations - the same reason we add empty handlers in a user command script. It should not break the generality, as it makes the difference between sending a signal only to the main container process or to its process group (YARN sends to a foreground group like bash does in the interactive mode). But yes, I'd rather used `exec "..."` in auto-generated Docker entry point in YARN without of `bash -c "..."` as it does not spawn an intermediate shell:

``` bash
exec /bin/bash -c "trap ' ' TERM; $JAVA_HOME/bin/java -DLAUNCHER_LOG_DIR=$LAUNCHER_LOG_DIR -Xmx256m com.microsoft.frameworklauncher.applicationmaster.Bootstrap 1>$LAUNCHER_LOG_DIR/stdout 2>$LAUNCHER_LOG_DIR/stderr "
```

So I'm open for better solutions. Without this handler, YARN task container just exits immediately.

Reply:
let's keep the default YARN behavior, otherwise, it is better to contribute "trap ' ' TERM" to YARN code and make a flag there.